### PR TITLE
feature: support for multiple web3 providers

### PR DESCRIPTION
**Pull request by lucasmenendez**
Explained in #58 

Changes:
* Update database schema and API endpoints to include the id of the chain of each token
* Delete chain id from service matadata on the database and remove the checks about the chain id
* Update the scanner and token related API endpoints to use the web3 provider associated to each token
* Handle a list of supported web3 providers as cmd/census3 entrypoint flag
* New endpoint (GET `/api/info`) to get api related information such as supported chain ids
* Include support for list of web3 providers on env file and docker-compose.yml

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # A web3 endpoint provider
-# WEB3_PROVIDER="https://mainnet.infura.io/v3/..."
-WEB3_PROVIDER="https://web3.dappnode.net"
+WEB3_PROVIDERS=https://eth.llamarpc.com,https://eth-goerli.api.onfinality.io/public
```
Comment (p4u) [inline .env:2]: Umm, maybe this should not go here? I don't think we have to deliver the code with a pre-configured and private web3 endpoint

File: state/providers.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	}
+
+	providers := make(map[int64]string)
+	for _, uri := range providersURIs {
+		cli, err := ethclient.Dial(uri)
+		if err != nil {
+			return nil, fmt.Errorf("error dialing web3 provider uri '%s': %w", uri, err)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (p4u) [inline state/providers.go:23]: Ideally these context timeout should be declared as constants.

Comment (lucasmenendez) [reply state/providers.go:23]: Yep, I have a task to review all the contexts timeouts

File: .env (hunk 1 above)
Comment (lucasmenendez) [reply .env:2]: I think that is a good practice to deliver the code with a template of configuration, we can replace those public endpoints with dummy urls

File: state/providers.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		defer cancel()
+		// get the chainID from the web3 endpoint
+		chainID, err := cli.ChainID(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("error getting the chainID from the web3 provider '%s': %w", uri, err)
+		}
+		providers[chainID.Int64()] = uri
```
Comment (p4u) [inline state/providers.go:30]: With this logic, if there is a web3 providers on the same chain declared twice, the last one will be the only stored.

A potential new feature that we might add in the future is to have a pool of web3 providers that automatically connect to the next if the current fail. You might take some ideas or code from the old ethservice queue we had when the Oracle was connecting to Ethereum (https://github.com/vocdoni/vocdoni-node/blob/release-azeno/service/ethevents.go#L109) and (https://github.com/vocdoni/vocdoni-node/blob/release-azeno/ethereum/handler/ethereumHandler.go#L90).

In any case, this is probably out of scope on this PR. However, we might start thinking about it; thus providers might be a `map [chainID] -> Web3ProviderHandler` 

Where Web3ProviderHandle has a list of methods, such as:

- AddProvider(string)
- NextProvider() ethClient
- GetProvider() ethClient
- String() string

The web3ProvidersHandler could make the Connect and check if the RPC endpoint actually works and is synced before returning it.

Sorry for the noise :P We might just open a new issue with this info.

**Review by p4u — APPROVED** [review]

File: state/providers.go (hunk 3 above)
Comment (lucasmenendez) [reply state/providers.go:30]: Created! https://github.com/vocdoni/census3/issues/73

**Pull request merged**